openbb_platform/core/openbb_core/api/router/commands.py

```diff
@@ -51,11 +51,14 @@ def build_new_signature(path: str, func: Callable) -> Signature:
     parameter_list = sig.parameters.values()
     return_annotation = sig.return_annotation
     new_parameter_list = []
-
-    for parameter in parameter_list:
+    var_kw_start = len(parameter_list)
+    for pos, parameter in enumerate(parameter_list):
         if parameter.name == "cc" and parameter.annotation == CommandContext:
             continue
 
+        if parameter.kind == Parameter.VAR_KEYWORD:
+            var_kw_start = pos
+
```

Comment:
"kwargs" is being forced as a "required" field in the body, and calling it VAR_KEYWORD or VAR_POSITIONAL is not allowed with a default value. `Optional[Dict]` is not actually optional without a default state. Setting as POSITIONAL_OR_KEYWORD allows you to set a default value of None, and **kwargs is still in the request body. 

It would be ideal if you didn't have to put kwargs in the body, but could just add them to the URL like every other parameter. This behaviour, though, is limited to the API. The Python interface does not require them to be declared within the field "kwargs".

```
        if parameter.name == "kwargs":
            new_parameter_list.append(
                Parameter(
                    parameter.name,
                    kind=parameter.POSITIONAL_OR_KEYWORD,
                    annotation=Optional[Dict[str, Any]],
                    default=None,
                )
            )
        else:
            new_parameter_list.append(
                Parameter(
                    parameter.name,
                    kind=parameter.kind,
                    default=parameter.default,
                    annotation=parameter.annotation,
                )
            )

Reply:
It's not possible to add undefined URL parameters to API endpoints

Reply:
Well then we can't get around that "nice to have" thing, but at least they can be unpacked the same as the Python interface.

This: `body = json.dumps({"data": data, "kwargs": {"chart_params": {"title": "my chart"}}})`

When unpacked becomes the same as the Python Interface "extras":

```
{'metadata': {'arguments': {'index': 'date',
   'lower_q': 0.25,
   'upper_q': 0.75,
   'model': 'std',
   'is_crypto': False,
   'trading_periods': None,
   'data': {'type': 'List[Data]',
    'columns': ['open',
     'vwap',
     'split_ratio',
     'close',
     'volume',
     'dividend',
     'date',
     'low',
     'high']},
   'chart_params': {'title': 'my chart'}},
  'duration': 109062167,
  'route': '/technical/cones',
  'timestamp': '2024-03-22T12:05:53.906124'}}
```

Reply:
VAR_KEYWORD parameters cannot have default values in Python, so they will be seen as mandatory by FastAPI.

URL parameters should be simple key value pairs and we use the body for more complex things

Reply:
Metadata reflects the call that was made to the api, if the call sends `'chart_params': {'title': 'my chart'}}` inside a parameter called `kwargs`/`some_kw_args`/`extra_params` it is expected that you see 

```
{'metadata': 
{'arguments': 
        {
         'index': 'date',
         'lower_q': 0.25,
         'data': {
                    'type': 'List[Data]',
                    'columns': ['open', ...],
          },
         'kwargs/some_kw_args/extra_params': {'chart_params': {'title': 'my chart'}}
         },
  'duration': 109062167,
  'route': '/technical/cones',
  'timestamp': '2024-03-22T12:05:53.906124'}}
```

Reply:
For this purpose, "extra_params" is a better name for this than "kwargs". Less confusion with everything else that is passed around as "kwargs".